xc/cancel_utils_test.ml

```diff
@@ -28,50 +26,27 @@ let xenstore_test xs _ =
     Thread.create
       (fun () ->
         Thread.delay 1. ;
-        Xenops_task.cancel tasks task.Xenops_task.id)
+        Xenops_task.with_cancel task (fun () -> ()) (fun () -> ()))
       ()
   in
   try
     let (_ : bool) =
       cancellable_watch (TestPath "/test/cancel") [] [] task ~xs ~timeout:3. ()
     in
     raise Did_not_cancel
-  with Cancelled _ -> (* success *)
-                      ()
-
-let subprocess_test _ =
-  let task = Xenops_task.add tasks "test" (fun _ -> None) in
-  let (_ : Thread.t) =
-    Thread.create
-      (fun () ->
-        Thread.delay 1. ;
-        Xenops_task.cancel tasks task.Xenops_task.id)
-      ()
-  in
-  try
-    let _, _ = cancellable_subprocess task "/bin/sleep" ["3s"] in
-    raise Did_not_cancel
-  with Cancelled _ -> (* success *)
-                      ()
+  with Xenopsd_error (Cancelled _) -> (* success *)
+                                      ()
 
 let _ =
   let verbose = ref false in
   Arg.parse
     [("-verbose", Arg.Unit (fun _ -> verbose := true), "Run in verbose mode")]
     (fun x -> Printf.fprintf stderr "Ignoring argument: %s\n" x)
     "Test cancellation functions" ;
-  (* We need xenstore anyway, so use this to verify that we're running in a
-     domain 0 environment *)
-  let xs =
-    try Xs.daemon_open ()
-    with _ ->
-      Printf.fprintf stderr
-        "Failed to open xenstore connection. This test suite must run in \
-         domain 0.\n" ;
-      exit 1
-  in
-  let suite =
-    "cancel test"
-    >::: ["subprocess" >:: subprocess_test; "xenstore" >:: xenstore_test xs]
-  in
-  run_test_tt ~verbose:!verbose suite
+  try
+    Xenstore.with_xs (fun xs ->
+        let suite = "cancel test" >::: ["xenstore" >:: xenstore_test xs] in
+        run_test_tt ~verbose:!verbose suite |> ignore)
+  with Xs_transport.Could_not_find_xenstore ->
+    (* ignore test, we're not running on domain 0 *)
```

Comment:
the behaviour has changed here? previously it looks like we ran `exit 1`

Reply:
That's correct, I don't want the test to fail because it's not running on dom0 as now it's run on every CI job.

There is still text being output when this connection fails, so it's not needed to add this here. This is only useful when testing this manually, I don't think the test was ever run in an automated way.

```
Failed to find xenstore socket. I tried the following:
  /var/run/xenstored/socket
  /proc/xen/xenbus
  /dev/xen/xenstore

On linux you might not have xenfs mounted:
   sudo mount -t xenfs xenfs /proc/xen
Or perhaps you just need to set the XENSTORED_PATH environment variable.
```

Reply:
Fair enough. I suppose we could use environment variables to detect whether we are running it in an automated way, but that might be too brittle